_docs/extensions/generate.py

```diff
@@ -0,0 +1,284 @@
+    with (ref_dir / 'libs-non-rel-raw.csv').open() as f:
+        non_rel_entries: list[_NonRelCSVRow] = list(csv.DictReader(f))  # type: ignore
+    non_rel_table = _get_non_relation_libs_table(non_rel_entries)
+    _write_if_needed(path=(gen_dir / 'libs-non-rel-table.rst'), content=non_rel_table)
+    # status key
+    key_table = _get_status_key_table()
+    msg = ' Library status is shown in the left column. See tooltips, or click here for a key.'
+    content = f'.. dropdown::{msg}\n\n' + '\n'.join('   ' + line for line in key_table.split('\n'))
+    _write_if_needed(path=(gen_dir / 'status-key-table.rst'), content=content)
+
+
+def _write_if_needed(path: pathlib.Path, content: str) -> None:
+    to_write = _FILE_HEADER + content
+    if not path.exists() or path.read_text() != to_write:
+        path.write_text(to_write)
+
+
+##########
+# tables #
+##########
+
+
+def _get_relation_libs_table(entries: list[_RelCSVRow]) -> str:
+    def key(row: tuple[str, ...]) -> tuple[str, ...]:
+        status, _name, _kind, desc = row
+        return status, desc
+
+    rows = [(_status(e), _name(e), _kind(e), _rel_description(e)) for e in entries]
+    return _LIBS_TABLE_HEADER + _rows_to_rst(sorted(rows, key=key))
+
+
+def _get_non_relation_libs_table(entries: list[_NonRelCSVRow]) -> str:
+    def key(row: tuple[str, ...]) -> tuple[str, ...]:
+        status, _name, kind, desc = row
+        return status, kind, desc
+
+    rows = [(_status(e), _name(e), _kind(e), _non_rel_description(e)) for e in entries]
+    return _LIBS_TABLE_HEADER + _rows_to_rst(sorted(rows, key=key))
+
+
+def _get_status_key_table() -> str:
+    rows = [
+        (_status({'status': s}), _STATUS_TOOLTIPS[s])  # type: ignore
+        for s in _STATUS_SORTKEYS
+        if s in _EMOJIS
+    ]
+    rows.append(('', 'None of the above.'))
+    return _KEY_TABLE_HEADER + _rows_to_rst(rows)
+
+
+##########
+# fields #
+##########
+
+
+def _status(entry: _CSVRow) -> str:
+    status = entry['status']
+    prefix = _hidden_text(_STATUS_SORTKEYS[status])
+    if status not in _EMOJIS:
+        return prefix
+    if status not in _STATUS_TOOLTIPS:
+        return f'{prefix}       | {_EMOJIS[status]}'
+    return f"""{prefix.rstrip()}
+          <div class="emoji-div">
+            {_EMOJIS[status]}
+            <div class="emoji-tooltip">{_STATUS_TOOLTIPS[status]}</div>
+          </div>
+
+"""
+
+
+def _name(entry: _CSVRow) -> str:
+    main_link = _rst_link(entry['name'], entry['url'])
+    extra_links = ', '.join([
+        _rst_link(_EMOJIS.get(text, '') + text, url)
+        for text in ('docs', 'src')
+        if (url := entry[text])
+    ])
+    if not extra_links:
+        return main_link
+    return f'{main_link} ({extra_links})'
+
+
+def _kind(entry: _CSVRow) -> str:
+    kind = entry['kind']
+    prefix = _hidden_text(_KIND_SORTKEYS[kind])
+    kind_str = _EMOJIS.get(kind, '') + kind
+    if not kind_str:
+        return prefix
+    return f'{prefix}       | {kind_str}'
```

Comment:
Is the number of spaces meaningful or important here?

Reply:
A comment would suffice for now.

Maybe a proper RST builder could be used in the future, a 3p library, or some custom code abstractions.